File: 18-frame-printer/frame_printer.py
# Python program to print words in a frame
# Written by M.V.Harish Kumar on 15-01-2024

s = input("Enter a string: ")
words = s.split()
max_len = max(len(word) for word in words)

print('*'*(max_len + 4))
for word in words:
    print(f"* {word:<{max_len}} *")
print('*'*(max_len + 4))

File: 18-frame-printer/frame_printer.rb
# Ruby program to print words in a frame
# Written by M.V.Harish Kumar on 01-02-2025

print "Enter a string: "
words = gets.chomp.split
mlen = words.map(&:length).max

puts '*' * (mlen+4)
words.each do |word|
  puts "* #{word.ljust mlen} *"
end
puts '*' * (mlen+4)
